### PR TITLE
dotnet-format check

### PR DESCRIPTION
**Pull request by eddynaka**
Partially addressing #957.

## Changes
- added dotnet-format in sanity workflow
- updated files with wrong format

For significant contributions please make sure you have completed the following items:

* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: OpenTelemetry.sln [hunk 1]
```diff
@@ -106,12 +106,14 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "ISSUE_TEMPLATE", "ISSUE_TEM
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "workflows", "workflows", "{E69578EB-B456-4062-A645-877CD964528B}"
 	ProjectSection(SolutionItems) = preProject
-		.github\workflows\docs.yml = .github\workflows\docs.yml
+		.github\workflows\docfx.yml = .github\workflows\docfx.yml
```
Comment (reyang) [inline OpenTelemetry.sln:109]: Thanks for the catch :)

Comment (eddynaka) [reply OpenTelemetry.sln:109]: thank you for the dotnet-format information 👍

File: .github/workflows/sanitycheck.yml [hunk 2]
```diff
@@ -29,3 +29,21 @@ jobs:
 
     - name: detect non-ASCII encoding and trailing space
       run: python3 ./build/sanitycheck.py
+
+  check-format:
```
Comment (reyang) [inline .github/workflows/sanitycheck.yml:33]: This one would probably worth its own workflow or get merged with other compilation/stylecop workflow.
Putting it here means the job would get triggered even for a PR that only touched Markdown files.

Not a blocker though.

**Comment by eddynaka** [comment]
> LGTM, we probably want to trigger this job only when there are changes on `*.cs` files or the style rule itself.

I'm changing to a specific workflow and i will add that rule

File: .github/workflows/sanitycheck.yml (hunk 2 above)
Comment (eddynaka) [reply .github/workflows/sanitycheck.yml]: just updated

**Comment by CodeBlanch** [comment]
This basically runs style cop or what?

**Comment by eddynaka** [comment]
> This basically runs style cop or what?

dotnet-format is a code formatter for dotnet that applies style preferences to a project or solution. Preferences will be read from an .editorconfig file, if present, otherwise a default set of preferences will be used. At this time dotnet-format is able to format C# and Visual Basic projects with a subset of supported .editorconfig options.

File: .github/workflows/dotnet-format.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+name: dotnet format
+
+on:
+  pull_request:
+    branches: [ master ]
+    paths:
+    - '**.cs'
```
Comment (reyang) [inline .github/workflows/dotnet-format.yml:7]: I guess we need to add `.editorconfig` as well in case the rule change caused failure.

Comment (eddynaka) [reply .github/workflows/dotnet-format.yml:7]: done

**Pull request merged**